### PR TITLE
SONAR-8402 Purge should delete queued task when deleting a project

### PR DESCRIPTION
**Pull request by simonbrandhof**

### CODE REVIEW

File: server/sonar-db-dao/src/test/java/org/sonar/db/purge/PurgeDaoTest.java [hunk 1]
```diff
@@ -155,6 +155,23 @@ public void delete_project_in_ce_activity_when_deleting_project() {
   }
 
   @Test
+  public void delete_tasks_in_ce_queue_when_deleting_project() {
+    ComponentDto projectToBeDeleted = dbTester.components().insertProject();
+    ComponentDto anotherLivingProject = dbTester.components().insertProject();
+
+    // Insert 2 rows in CE_ACTIVITY : one for the project that will be deleted, and on on another project
```
Comment (sns-seb) [inline server/sonar-db-dao/src/test/java/org/sonar/db/purge/PurgeDaoTest.java]: c/p error: CE_QUEUE
typo: "on on another project" => "one on another project"

File: server/sonar-db-dao/src/test/java/org/sonar/db/purge/PurgeDaoTest.java [hunk 2]
```diff
@@ -222,13 +239,19 @@ public void deleteProject_deletes_webhook_deliveries() {
...
-  private CeActivityDto insertCeActivity(String componentUuid) {
+  private CeQueueDto createCeQueue(ComponentDto component) {
     CeQueueDto queueDto = new CeQueueDto();
     queueDto.setUuid(Uuids.create());
     queueDto.setTaskType(REPORT);
-    queueDto.setComponentUuid(componentUuid);
+    queueDto.setComponentUuid(component.uuid());
     queueDto.setSubmitterLogin("henri");
     queueDto.setCreatedAt(1_300_000_000_000L);
+    queueDto.setStatus(CeQueueDto.Status.PENDING);
```
Comment (sns-seb) [inline server/sonar-db-dao/src/test/java/org/sonar/db/purge/PurgeDaoTest.java]: you should randomly pick a `CeQueueDto.Status` value, that would enforce that the value doesn't matter

File: server/sonar-db-dao/src/test/java/org/sonar/db/purge/PurgeDaoTest.java [hunk 3]
```diff
@@ -222,13 +239,19 @@ public void deleteProject_deletes_webhook_deliveries() {
...
+    return queueDto;
+  }
+
+  private CeActivityDto insertCeActivity(ComponentDto component) {
+    CeQueueDto queueDto = createCeQueue(component);
 
     CeActivityDto dto = new CeActivityDto(queueDto);
     dto.setStatus(CeActivityDto.Status.SUCCESS);
```
Comment (sns-seb) [inline server/sonar-db-dao/src/test/java/org/sonar/db/purge/PurgeDaoTest.java]: legacy code, but same comment as for CeQueueDto#status above

File: server/sonar-db-dao/src/test/java/org/sonar/db/purge/PurgeDaoTest.java (hunk 2 above)
Comment (simonbrandhof) [inline server/sonar-db-dao/src/test/java/org/sonar/db/purge/PurgeDaoTest.java]: I changed the test in order to test both status.

**Pull request closed without merging**